src/lib/storage/fake_device_factory.rb

```diff
@@ -242,7 +242,7 @@ def create_partition_table(parent, args)
       # @return [Fixnum]
       def str_to_ptable_type(string)
         # Allow different spelling
-        string = "msdos" if string.downcase == "ms-dos"
+        string = "msdos" if string.casecmp("ms-dos").zero?
```

Comment:
I don't have a strong opinion either. For me `casecmp` is acceptable, although obviously less readable. In this case the performance gain is marginal but maybe in some other scenario with a big loop or something...

@jreidinger suggests to write a generic helper. For that kind of things we should use add another extension to `Yast::CoreExt`[1]. Let's say `Yast::CoreExt::ComparableString` for example. But let's not forget that this change is suggested because of performance reasons and overloading String is not free performance-wise. So we could end up with a whole mechanism to bring readability back that actually ends up killing the original reason to decrease readability in the first place. :wink:

So in short, I'm fine with the change to start using `casecmp`.  I'm also fine with adding a generic helper to abstract that, but only if we can prove that is worth it from the performance POV.

[1] https://github.com/yast/yast-ruby-bindings/tree/master/src/ruby/yast/core_ext